PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetExtractor.cxx

```diff
@@ -857,20 +827,27 @@ Bool_t AliAnalysisTaskJetExtractor::Run()
       GetJetType(jet, currentJetType_HM, currentJetType_PM, currentJetType_IC);
       // Get true estimators: for pt, jet mass, ...
       GetTrueJetPtFraction(jet, truePtFraction, truePtFraction_PartLevel);
-      // Here we can use two methods for matching which appear in the following order
-      // (1) Use the methodology of the tagger for matching. (Note ONLY pTs should be filled not distances etc.)
-      // (2) Use the naive matching, which is strictly geometric.
-      // Controlling which option for matching is done with the variable fUseTaggerForMatching
-      if(fUseTaggerForMatching){
-	GetJetTagging(jet, *jetsHybrid, *jetsDetLevel, *jetsPartLevel, matchedJetPt_Det, matchedJetPt_Part);
+      
+      // Get the Matched Observables
+      AliJetContainer * jetsHybrid = GetJetContainer(0);
+      AliJetContainer * jetsDetLevel = GetJetContainer(1);
+      AliJetContainer * jetsPartLevel = GetJetContainer(2);
+      // Validation                                                                                                                                                                                      
+      if (!jetsHybrid) {
+	AliErrorStream() << "Could not retrieve hybrid jet collection.\n";
+	return kFALSE;
       }
-      else{
-	if(fMatchedDetLevelJetsArrayName != "")
-	  GetMatchedJetObservables(jet, matchedJetPt_Det, matchedJetDistance_Det, matchedJetMass_Det, matchedJetAngularity_Det, matchedJetpTD_Det, kFALSE);
-	// if we have a second matching array name given, do the second Matching
-	if(fMatchedPartLevelJetsArrayName != "")
-	  GetMatchedJetObservables(jet, matchedJetPt_Part, matchedJetDistance_Part, matchedJetMass_Part, matchedJetAngularity_Part, matchedJetpTD_Part, kTRUE);
+      if (!jetsDetLevel) {
+	AliErrorStream() << "Could not retrieve det level jet collection.\n";
+	return kFALSE;
       }
+      if (fDoPartLevelMatching && !jetsPartLevel) {
+	AliErrorStream() << "Could not retrieve part level jet collection.\n";
+	return kFALSE;
+      }
+      GetMatchedJetObservables(jet, *jetsHybrid, *jetsDetLevel, *jetsPartLevel,  matchedJetPt_Det, matchedJetPt_Part, matchedJetDistance_Det, matchedJetDistance_Part, matchedJetMass_Det, matchedJetMass_Part, matchedJetAngularity_Det, matchedJetAngularity_Part, matchedJetpTD_Det,matchedJetpTD_Part);
```

Comment:
I think this can still be a bit thinned out. There is no need to define the containers the containers jetsHybrid, jetsDetLevel, jetsPartLevel here. They could be defined in GetMatchedJetObservables() so you would not need to pass them to the function.

Also, I think you don't need to check whether the jet containers are valid for each jet. Once in the beginning of the task should suffice. There you could also activate fDoPartLevelMatching & fDoDetLevelMatching, depending on which container is present.

Reply:
Sure I can change this.